core/src/main/java/org/elasticsearch/index/fielddata/IndexFieldDataService.java

```diff
@@ -64,9 +61,17 @@
     public static final String FIELDDATA_CACHE_KEY = "index.fielddata.cache";
     public static final String FIELDDATA_CACHE_VALUE_NODE = "node";
 
+    private static final IndexFieldData.Builder MISSING_DOC_VALUES_BUILDER = new IndexFieldData.Builder() {
+        @Override
+        public IndexFieldData<?> build(Index index, Settings indexSettings, MappedFieldType fieldType, IndexFieldDataCache cache, CircuitBreakerService breakerService, MapperService mapperService) {
+            throw new IllegalStateException("Fielddata can't be loaded on field [" + fieldType.names().fullName()
```

Comment:
I'm not sure what this message means - maybe better to say that fielddata is unsupported for these types and to use docvalues instead?